### PR TITLE
Bucket & Function & Passport Feat and  Fix

### PR DESCRIPTION
**Pull request by tthiscafer**

### CODE REVIEW

File: stacks/spica/src/bucket/services/bucket.initializer.ts [hunk 1]
```diff
@@ -12,17 +12,19 @@ export class BucketInitializer {
...
-            category: RouteCategory.Content,
-            id: `bucket_${bucket._id}`,
-            icon: bucket.icon,
-            path: `/bucket/${bucket._id}`,
-            display: bucket.title
-          })
-        );
-      });
+      buckets
+        .sort((r1, r2) => r1.order - r2.order)
```
Comment (thesayyn) [inline stacks/spica/src/bucket/services/bucket.initializer.ts]: No need for this because of the data comes ordered from api. Please revert this change

File: stacks/spica/src/bucket/services/bucket.service.ts [hunk 2]
```diff
@@ -53,6 +53,9 @@ export class BucketService {
       .pipe(tap(b => this.store.dispatch(new fromBucket.Upsert(b))));
   }
 
+  updateMany(buckets: Bucket[]): Observable<Bucket[]> {
+    return this.http.put<Bucket[]>(`api:/bucket/order`, buckets);
```
Comment (thesayyn) [inline stacks/spica/src/bucket/services/bucket.service.ts]: Please remove order suffix

File: stacks/spica/src/function/pages/add/add.component.html [hunk 3]
```diff
@@ -114,15 +114,15 @@ <h4 mat-subheader>
           </button>
         </h4>
         <mat-list-item class="trigger" *ngFor="let trigger of function.triggers | keyvalue">
-          <mat-form-field>
+          <mat-form-field class="trigger-form-field">
```
Comment (thesayyn) [inline stacks/spica/src/function/pages/add/add.component.html]: Consider `mat-form-field.trigger` selector

File: stacks/spica/src/function/pages/add/add.component.html [hunk 4]
```diff
@@ -114,15 +114,15 @@ <h4 mat-subheader>
...
             <mat-label>Handler</mat-label>
             <mat-select [disabled]="true" value="trigger.key">
               <mat-option value="trigger.key">{{ trigger.key }}</mat-option>
             </mat-select>
           </mat-form-field>
 
           <ng-container *ngIf="triggers | async; let triggers">
-            <mat-form-field>
+            <mat-form-field class="trigger-form-field">
```
Comment (thesayyn) [inline stacks/spica/src/function/pages/add/add.component.html]: Consider `mat-form-field.trigger` selector

File: stacks/spica/src/function/pages/add/add.component.html [hunk 5]
```diff
@@ -211,10 +212,10 @@ <h4 mat-subheader>
           </button>
         </h4>
         <mat-list-item environment *ngFor="let env of function.env; let index = index">
-          <mat-form-field>
+          <mat-form-field class="trigger-form-field">
```
Comment (thesayyn) [inline stacks/spica/src/function/pages/add/add.component.html]: Consider `mat-form-field.trigger` selector

File: stacks/spica/src/function/pages/add/add.component.html [hunk 6]
```diff
@@ -211,10 +212,10 @@ <h4 mat-subheader>
...
             <input matInput placeholder="Name" [(ngModel)]="env.name" />
           </mat-form-field>
-          <mat-form-field>
+          <mat-form-field class="trigger-form-field">
```
Comment (thesayyn) [inline stacks/spica/src/function/pages/add/add.component.html]: Consider `mat-form-field.trigger` selector

File: stacks/spica/src/function/pages/add/add.component.html [hunk 7]
```diff
@@ -143,6 +143,7 @@ <h4 mat-subheader>
 
             <ng-container ngModelGroup="triggerOptions">
               <span
+                class="trigger-margin"
```
Comment (thesayyn) [inline stacks/spica/src/function/pages/add/add.component.html]: Consider `mat-form-field.margin` selector

File: stacks/spica/src/function/pages/add/add.component.scss [hunk 8]
```diff
@@ -1,4 +1,18 @@
 :host ::ng-deep {
+  .trigger-form-field,
+  .trigger-margin {
+    margin: 10px 10px 0 0;
+  }
+  .mat-subheader {
+    padding: 0;
+  }
+  mat-divider {
+    // left: 5% !important;
```
Comment (thesayyn) [inline stacks/spica/src/function/pages/add/add.component.scss]: Remove unused code

File: stacks/spica/src/function/pages/add/add.component.scss [hunk 9]
```diff
@@ -32,21 +46,29 @@
...
         height: auto;
-        padding: 15px 0px;
+
         .mat-list-item-content {
           display: contents;
         }
       }
       mat-list {
         .editor {
+          .decorationsOverviewRuler {
```
Comment (thesayyn) [inline stacks/spica/src/function/pages/add/add.component.scss]: Use dash-cased  class names

File: stacks/spica/src/function/pages/subscription-index/subscription-index.component.html [hunk 10]
```diff
@@ -25,8 +25,8 @@ <h4 class="mat-h4">
       </ng-container>
 
       <ng-container matColumnDef="actions">
-        <mat-header-cell *matHeaderCellDef>Actions</mat-header-cell>
-        <mat-cell *matCellDef="let element">
+        <mat-header-cell class="action-cell" *matHeaderCellDef>Actions</mat-header-cell>
```
Comment (thesayyn) [inline stacks/spica/src/function/pages/subscription-index/subscription-index.component.html]: Consider `mat-header-cell.action` selector

File: stacks/spica/src/function/pages/subscription-index/subscription-index.component.html [hunk 11]
```diff
@@ -25,8 +25,8 @@ <h4 class="mat-h4">
...
+        <mat-cell class="action-cell" *matCellDef="let element">
```
Comment (thesayyn) [inline stacks/spica/src/function/pages/subscription-index/subscription-index.component.html]: Consider `mat-header-cell.action` selector

Comment (tthiscafer) [reply stacks/spica/src/function/pages/subscription-index/subscription-index.component.html]: This class is not my custom class. This class using by Angular Material.
